MonMaraicher/Features/Map/MapViewModel.swift

```diff
@@ -42,6 +41,53 @@ final class MapViewModel: ObservableObject {
         CLLocationManager().requestWhenInUseAuthorization()
         CLLocationManager().desiredAccuracy = kCLLocationAccuracyBest
     }
+
+    func onNearbyFarmerButtonTapped() {
+        do {
+            if let nearbyFarmer = try searchNearbyFarmer() {
+                let nearbyFarmerCoordinate = CLLocationCoordinate2D(latitude: nearbyFarmer.location.latitude,
+                                                                    longitude: nearbyFarmer.location.longitude)
+                let nearbyFarmerRegion = MKCoordinateRegion(center: nearbyFarmerCoordinate,
+                                                            span: self.nearbyFarmerSpan)
+                mapCameraPosition = .region(nearbyFarmerRegion)
+            }
+        } catch {
+            print("Aucun maraîcher trouvé à proximité")
+        }
+    }
+
+    private func searchNearbyFarmer() throws -> Farmer? {
```

Comment:
Why is your method marked with 'throws'? You don't throw inside...

Reply:
Yes, I'm trying to set it up